wolfssl/openssl/ed25519.h

```diff
@@ -28,6 +28,26 @@
 extern "C" {
 #endif
 
+#ifndef WOLFSSL_ED25519_TYPE_DEFINED /* guard on redeclaration */
+typedef struct WOLFSSL_ED25519   WOLFSSL_ED25519;
+#define WOLFSSL_ED25519_TYPE_DEFINED
+#endif
+
+typedef struct WOLFSSL_ED25519 {
+    WOLFSSL_BIGNUM* p; /* public key */
+    WOLFSSL_BIGNUM* k; /* private key */
+    void* internal;
+    /* bits */
+    byte inSet:1;     /* internal set from external ? */
+    byte exSet:1;     /* external set from internal ? */
+} WOLFSSL_ED25519;
```

Comment:
You are `typedef`'ing the struct twice. In effect the above redeclaration guard doesn't actually do anything. I think the compiler doesn't complain since its being `typedef`'d the same way twice.